docs/querying/api.md

```diff
@@ -363,6 +363,85 @@ $ curl http://localhost:9090/api/v1/targets
 }
 ```
 
+## Target metadata
+
+The following endpoint returns metadata about metrics currently scraped by targets.
+
+```
+GET /api/v1/targets/metadata
+```
+
+URL query parameters:
+
+- `match=<series_selector>`: A selector that matches targets by label matchers. If a metric name is provided, only metadata for that metric name is returned.
```

Comment:
You should be clear about what the label matchers are matching against

I'm wondering now if we should split the metric name from the target labels URL parameter wise, it's a bit unclean to have them as one

Reply:
Fine with me. `metric` and `match_target` ok? `target_matcher` would work too but be inconsistent with our other endpoints with match params.

Reply:
sgtm